### DISCUSSION TITLE
Got error of `Resource not accessible by integration` where re-creating the same name repo

### DISCUSSION THREAD

**Discussion by ssrlive** [community/community#60820] [Actions] [answered]
### Select Topic Area

Question

### Body

I have deleted [my repo](https://github.com/ssrlive/socks5-impl) and re-created it with the same name.
In my github action file exists this code
```yml
      - name: Upload
        uses: softprops/action-gh-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          files: release/*
```
to deploy my release package.

But I got the error this time
```
⚠️ Unexpected error fetching GitHub release for tag refs/tags/v0.3.7: HttpError: Resource not accessible by integration
Error: Resource not accessible by integration
```

How I resolve this issue?

**Comment by the-officialjosh** [comment, marked as answer]
I am not sure, but I guess the error message suggests that the integration (GitHub Actions) cannot access the resource (GitHub release) for the repository you recreated with the same name.

When you delete a repository and recreate it with the same name, it is possible that some settings or permissions related to the integration (GitHub Actions) are not properly updated. To resolve this issue, you can try the following steps:

1. Disable and re-enable the integration: Go to the repository's settings, navigate to the "Integrations & services" (or similar) section, find the GitHub Actions integration, and disable it. Then, re-enable it again. This can help refresh the integration's access to the repository.

2. Check repository permissions: Ensure that the GitHub Actions integration has the necessary permissions to access the repository and perform the required actions. Make sure it has appropriate read and write access.

3. Review workflow configuration: Double-check your workflow file to ensure that it references the correct repository name and settings. Confirm that the release package and tag you are trying to access are properly set up in the workflow.

4. Verify authentication: Ensure that the `GITHUB_TOKEN` used in your workflow file is correctly set and has the necessary permissions to access and modify releases.

If the issue persists after following these steps, consider reaching out to GitHub support for further assistance. They can provide more specific guidance based on your situation and help resolve the issue with the resource accessibility.

**Reply by josealmir** [reply]
Very good!
Apply these passes to my repository and fix my action.

**Reply by charlesdzadu** [reply]
Work for me.
I just have to make sure Github Actions have read and write permissions.

**Reply by userddssilva** [reply]
Thanks for this tips, it works for me!

**Reply by dangthanhvinh003** [reply]
Work for me.

**Reply by niteshraj2310** [reply]
Thanks a ton!!!

**Comment by ssrlive** [comment]
Solved. Thank you.

![image](https://github.com/community/community/assets/30760636/09ad3c38-a26c-45e0-b681-c9f8eda616b3)

**Reply by fujohnwang** [reply]
same same

**Reply by michaelhays** [reply]
I converted my organization into an enterprise, and it looks like doing so set the default setting across the enterprise to "Read repository contents and packages permissions".

You can go to https://github.com/enterprises/{enterprise_name}/settings/actions (replace `{enterprise_name}`) to update this.

**Reply by HyMike** [reply]
this worked! thank you.

**Reply by MECR-0209** [reply]
dawg I was trying to put my wakatime statistics on my profile and after 2 days and 5 hours I got this solution.
You saved me thank you

**Reply by dyspop** [reply]
This worked for me. Thanks!

**Reply by eruvanos** [reply]
Better add
```
permissions:
  contents: write
```

**Reply by younsl** [reply]
If you get a "Resource not accessible by integration" error with the [8398a7/action-slack](https://github.com/8398a7/action-slack) step, you can fix it by adding:

```yaml
jobs:
  my-job:
    # As a best practice, it's safer to set permissions at the job level
    # rather than globally for the entire workflow.
    permissions:
      actions: read
      contents: read
```

**Comment by ozbillwang** [comment]
add this in top of your action file

```
permissions: write-all
```

**Reply by hraban** [reply]
This is the simplest answer.

More information: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

E.g. to allow your workflow to create a new release:

```
permissions:
  contents: write
```

**Reply by snail2sky** [reply]
Thank you very much, it really works

**Comment by braden-w** [comment]
If anyone is here for tag-based workflows, the solution for me was simple: I had created a git tag locally that was ahead of my remote main branch. The tauri-action was trying to create a release for a commit that didn't exist on the remote repository.

Fix: Simply push your local main branch to origin before creating/pushing tags:

```bash
git push origin main
git push origin v1.0.0  # or whatever your tag is
```

This ensures the commit that the tag points to actually exists on the remote repository when tauri-action tries to create the release.
